app/src/main/java/com/example/meetup/Fragments/ComposeFragment.java

```diff
@@ -0,0 +1,205 @@
+
+            }
+        });
+        submitCompose.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String description = descriptionCompose.getText().toString();
+                String caption = captionCompose.getText().toString();
+                String category = categoryCompose.getText().toString();
+                String startupName = startupNameCompose.getText().toString();
+                if (description.isEmpty()) {
+                    Toast.makeText(getContext(), "description can't be empty", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                if (photoFile == null) {
+                    Toast.makeText(getContext(), "there is no image", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                ParseUser currentUser = ParseUser.getCurrentUser();
+                savePost(startupName, description, caption, category, currentUser, photoFile);
+            }
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        //Detects request codes
+        if(requestCode==GET_FROM_GALLERY && resultCode == Activity.RESULT_OK) {
+            Uri selectedImage = data.getData();
+            Bitmap pictureBitmap = null;
+            try {
+                pictureBitmap = MediaStore.Images.Media.getBitmap(getContext().getContentResolver(), selectedImage);
+            } catch (FileNotFoundException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+            photoFile = bitmapToFile(getContext(), pictureBitmap, "image.jpeg");
+        }
+    }
+
+    public static File bitmapToFile(Context context, Bitmap bitmap, String fileNameToSave) { // File name like "image.png"
+        //create a file to write bitmap data
+        File file = null;
+        try {
+            // use cache to store image
+            file = new File(context.getCacheDir(), fileNameToSave);
+
+            file.createNewFile();
+
+            //Convert bitmap to byte array
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            bitmap.compress(Bitmap.CompressFormat.PNG, 0 , bos); // YOU can also save it in JPEG
+            byte[] bitmapdata = bos.toByteArray();
+
+            //write the bytes in file
+            FileOutputStream fos = new FileOutputStream(file);
+            fos.write(bitmapdata);
+            fos.flush();
+            fos.close();
+            return file;
+        }catch (Exception e){
+            e.printStackTrace();
+            return file; // it will return null
+        }
```

Comment:
We can use `finally` to return the file or to execute any lines that need to be run whether there's an exception or not.
```java
try {
  ...
  FileOutputStream fos = new FileOutputStream(file);
  fos.write(bitmapdata);
  fos.flush();
  fos.close();
} catch (Exception e) {
  e.printStackTrace();
} finally{
  return file; 
}
```